core/rawdb/freezer.go

```diff
@@ -285,12 +291,42 @@ func (f *freezer) TruncateAncients(items uint64) error {
 	if atomic.LoadUint64(&f.frozen) <= items {
 		return nil
 	}
+	var frozen uint64
+	for _, table := range f.tables {
+		if err := table.truncateHead(items); err != nil {
+			return err
+		}
+		// Tables should be aligned, only check the first table.
+		if frozen == 0 {
+			frozen = atomic.LoadUint64(&table.items)
+		}
+	}
+	atomic.StoreUint64(&f.frozen, frozen)
+	return nil
+}
+
+// TruncateTail discards any recent data below the provided threshold number.
+func (f *freezer) TruncateTail(tail uint64) error {
+	if f.readonly {
+		return errReadOnly
+	}
+	f.writeLock.Lock()
+	defer f.writeLock.Unlock()
+
+	if atomic.LoadUint64(&f.tail) >= tail {
+		return nil
+	}
+	var truncated uint64
 	for _, table := range f.tables {
-		if err := table.truncate(items); err != nil {
+		if err := table.truncateTail(tail); err != nil {
 			return err
 		}
+		if truncated == 0 {
+			// Tables should be aligned, only check the first table.
```

Comment:
Same here

Reply:
Fixed, this is supposed to be deleted but I forgot to do it. Thanks for catching it!